newIDE/app/src/Utils/GDevelopServices/Game.js

```diff
@@ -287,120 +315,172 @@ export const setGameUserAcls = (
-        headers: {
-          Authorization: authorizationHeader,
-        },
-      })
-    )
-    .then(response => response.data);
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.get(
+    `${GDevelopGameApi.baseUrl}/game/${gameId}`,
+    {
+      params: {
+        userId,
+      },
+      headers: {
+        Authorization: authorizationHeader,
+      },
+    }
+  );
+
+  return response.data;
 };
 
-export const deleteGame = (
+export const deleteGame = async (
   getAuthorizationHeader: () => Promise<string>,
   userId: string,
   gameId: string
 ): Promise<Game> => {
-  return getAuthorizationHeader()
-    .then(authorizationHeader =>
-      axios.delete(`${GDevelopGameApi.baseUrl}/game/${gameId}`, {
-        params: {
-          userId,
-        },
-        headers: {
-          Authorization: authorizationHeader,
-        },
-      })
-    )
-    .then(response => response.data);
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.delete(
+    `${GDevelopGameApi.baseUrl}/game/${gameId}`,
+    {
+      params: {
+        userId,
+      },
+      headers: {
+        Authorization: authorizationHeader,
+      },
+    }
+  );
+  return response.data;
 };
 
-export const getGames = (
+export const getGames = async (
   getAuthorizationHeader: () => Promise<string>,
   userId: string
 ): Promise<Array<Game>> => {
-  return getAuthorizationHeader()
-    .then(authorizationHeader =>
-      axios.get(`${GDevelopGameApi.baseUrl}/game`, {
-        params: {
-          userId,
-        },
-        headers: {
-          Authorization: authorizationHeader,
-        },
-      })
-    )
-    .then(response => response.data);
+  const authorizationHeader = await getAuthorizationHeader();
+
+  const response = await axios.get(`${GDevelopGameApi.baseUrl}/game`, {
+    params: {
+      userId,
+    },
+    headers: {
+      Authorization: authorizationHeader,
+    },
+  });
+
+  return response.data;
+};
+
+export const getPublicGame = async (gameId: string): Promise<PublicGame> => {
+  const response = await axios.get(
+    `${GDevelopGameApi.baseUrl}/public-game/${gameId}`
+  );
+  return response.data;
+};
+
+export const getGameCategories = async (): Promise<GameCategory[]> => {
+  const response = await axios.get(`${GDevelopGameApi.baseUrl}/game-category`);
+  return response.data;
 };
 
-export const getPublicGame = (gameId: string): Promise<PublicGame> => {
-  return axios
-    .get(`${GDevelopGameApi.baseUrl}/public-game/${gameId}`)
-    .then(response => response.data);
+export const buyGameFeaturing = async (
+  getAuthorizationHeader: () => Promise<string>,
+  {
+    gameId,
+    userId,
+    usageType,
+  }: {| gameId: string, userId: string, usageType: GameUsageType |}
+): Promise<void> => {
+  const authorizationHeader = await getAuthorizationHeader();
+  await axios.post(
+    `${GDevelopGameApi.baseUrl}/game/action/buy-with-credits`,
+    {},
```

Comment:
A POST with no body?

Reply:
indeed, the few params are in the url. (we have other endpoints like this for "actions" mainly)